Controller/Payment/Order.php

```diff
@@ -249,10 +249,19 @@ public function execute()
                                     . " and amount:" . $amount);
             // @codeCoverageIgnoreEnd
 
+            $currency = $mazeOrder->getOrderCurrencyCode();
+
+            if ($currency === "KWD" or
+                $currency === "OMR" or
+                $currency === "BHD")
+            {
+                throw new \Exception($currency . " currency is not supported at the moment.");
```

Comment:
@yashgit891 , please return response same like error so that it can be used. No need to throw the exception. Also log it.

Reply:
same like this one 
```$responseContent = [
                'message'   => 'Unable to create your order. Please contact support.',
                'parameters' => []
            ];
```